### PR TITLE
Viem provider

### PR DESCRIPTION
**Pull request by yoanslvy**

### CODE REVIEW

File: src/wallet/viem/ViemAccount.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  PrivateKeyAccount,
+  SignableMessage,
+  toHex
+} from 'viem'
+import { toAccount } from 'viem/accounts'
+
+import { ComethWallet } from '../ComethWallet'
+
+export const getViemAccount = async (
+  ComethWallet: ComethWallet
```
Comment (shoto290) [inline src/wallet/viem/ViemAccount.ts:14]: Bad name "ComethWallet" -> "comethWallet"

File: src/wallet/viem/ViemAccount.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      return `0x${signedMessage}`
+    },
+    async signTransaction(transaction): Promise<any> {
+      const safeTxDataTyped = await ComethWallet.buildTransaction({
+        to: transaction.to as string,
+        value: toHex(transaction.value as bigint),
+        data: transaction.data as string
+      })
+
+      return await ComethWallet.signTransaction(safeTxDataTyped)
```
Comment (shoto290) [inline src/wallet/viem/ViemAccount.ts:34]: await not required

File: src/wallet/viem/ViemAccount.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    address,
+    async signMessage({
+      message
+    }: {
+      message: SignableMessage
+    }): Promise<Hash> {
+      const signedMessage = await ComethWallet.signMessage(message.toString())
+      return `0x${signedMessage}`
+    },
+    async signTransaction(transaction): Promise<any> {
```
Comment (shoto290) [inline src/wallet/viem/ViemAccount.ts:27]: any?

File: src/wallet/viem/ViemClient.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  avalanche,
+  avalancheFuji,
+  gnosis,
+  gnosisChiado,
+  polygonZkEvm,
+  polygonZkEvmTestnet,
+  musterTestnet
+]
+
+export const ViemClient = async (
```
Comment (shoto290) [inline src/wallet/viem/ViemClient.ts:36]: name

File: src/wallet/viem/customActions/getTransaction.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  return filteredTransactionEvent
+}
+
+export const getTransaction = async (
+  client: PublicClient<Transport, Chain>,
+  wallet: ComethWallet,
+  safeTxHash: Hash
+): Promise<TransactionReceipt> => {
+  const currentBlockNumber = await client.getBlockNumber()
+  const from = (await wallet.getAddress()) as Address
```
Comment (shoto290) [inline src/wallet/viem/customActions/getTransaction.ts:62]: await not needed

**Review by shoto290 — APPROVED** [review]

**Review by fdematos — APPROVED** [review]

**Pull request merged**